### PR TITLE
fix reboot and poweroff behavior. support poweroff(at)

### PR DESCRIPTION
**Pull request by shigechika**
I fixed reboot() and poweroff() behavior:
1. in_min
1. at

This logic is same as the ansible.
And added supports poweroff(at).
@rsmekala, Thank you for your advices.

Thank you.

### CODE REVIEW

**Comment by jnpr-community-netdev** [comment]
Can one of the admins verify this patch?

File: lib/jnpr/junos/utils/sw.py [hunk 1]
```diff
@@ -986,7 +986,7 @@ def reboot(self, in_min=0, at=None, all_re=True, on_node=None):
                     cmd.append(E('both-routing-engines'))
                 elif self._mixed_VC is True:
                     cmd.append(E('all-members'))
-        if in_min >= 0 and at is None:
+        if in_min >= 0:
```
Comment (rsmekala) [inline lib/jnpr/junos/utils/sw.py]: The default value of `in_min` is 0. So this condition is always true.

The test case is failing due to this.

A better logic would be 
```
# in_min given
if in_min > 0:
    append in

# in_min not given but at given
elif at is not None:
    append at

# default case
else:
    apppend in = 0
```

Comment (shigechika) [reply lib/jnpr/junos/utils/sw.py]: It's good idea!

**Comment by shigechika** [comment]
Please verify again.

**Comment by jnpr-community-netdev** [comment]
Can one of the admins verify this patch?

**Comment by shigechika** [comment]
I am using this patch since several months ago.
Of course, works fine.
-- Shigechika AIKAWA

**Comment by vnitinv** [comment]
Sorry for getting late on this @shigechika 
How about changing this

```
        if in_min > 0:
            # in_min given
            cmd.append(E('in', str(in_min)))
        elif at is not None:
            # in_min not given but at given
            cmd.append(E('at', str(at)))
        else:
            # default case (in_min=0)
            cmd.append(E('in', str(in_min)))
```
to

```
        if in_min >= 0:
            # in_min given or default case
            cmd.append(E('in', str(in_min)))
        elif at is not None:
            # in_min not given but at given
            cmd.append(E('at', str(at)))
```

**Comment by rsmekala** [comment]
@shigechika 'at' and 'in' should be mutually exclusive options in Junos. Tracked via PR# 1461215

**Comment by rsmekala** [comment]
ok to test

**Comment by rsmekala** [comment]
@shigechika Thanks for the contribution. Merging the pull request. ✌️

**Pull request merged**